File: backend/src/main/java/com/festago/dto/FestivalResponse.java
package com.festago.dto;

import com.festago.domain.Festival;
import java.time.LocalDate;

public record FestivalResponse(
    Long id,
    String name,
    LocalDate startDate,
    LocalDate endDate,
    String thumbnail) {

    public static FestivalResponse from(Festival festival) {
        return new FestivalResponse(
            festival.getId(),
            festival.getName(),
            festival.getStartDate(),
            festival.getEndDate(),
            festival.getThumbnail());
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/FestivalResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.domain.model.Festival
import kotlinx.serialization.Serializable
import java.time.LocalDate

@Serializable
data class FestivalResponse(
    val id: Int,
    val name: String,
    val startDate: String,
    val endDate: String,
    val thumbnail: String,
) {
    fun toDomain() = Festival(
        id = id.toLong(),
        name = name,
        startDate = LocalDate.parse(startDate),
        endDate = LocalDate.parse(endDate),
        thumbnail = thumbnail,
    )
}
